default.nix

```diff
@@ -20,10 +23,23 @@ let
   #       };
   #     "other-package": { ... };
   #   }
-  snapshotFromPackageLockJson = packageLockJson:
+  snapshotFromPackageLockJson = packageLockJson: pname: version:
     let
       packageLock = builtins.fromJSON (builtins.readFile packageLockJson);
 
+      # Load custom name and version of the program in case it was specified and
+      # not specified by the package-lock.json
+      topPackageName =
+          if builtins.hasAttr "name" packageLock then packageLock.name
+          else if ! isNull pname then pname
+          else fallbackPackageName;
+
+      updateTopPackageVersion = obj:
+        if builtins.hasAttr "version" obj then obj
+        else obj // {
+          version = if isNull version then fallbackPackageVersion else version;
```

Comment:
I might be going blind here, but where does `version` come from?

Reply:
`version` is a new argument for `snapshotFromPackageLockJson`. I also added `version` argument to the `buildPackage` builder and then propagated it into `snapshotFromPackageLockJson`. As a consequence, `snapshotFromPackageLockJson ` now takes two additional arguments (`pname` and `version`), but I hope it is not a problem as the only public API is `buildPackage` from what I understand.

Reply:
```suggestion
          version = ifNotNull version fallbackPackageVersion;
```